WooCommerce/Classes/ViewRelated/Products/Edit Product/Linked Products List Selector/LinkedProductListSelectorDataSource.swift

```diff
@@ -70,21 +77,22 @@ final class GroupedProductListSelectorDataSource: PaginatedListSelectorDataSourc
         cell.update(viewModel: viewModel, imageService: imageService)
 
         cell.configureAccessoryDeleteButton { [weak self] in
-            ServiceLocator.analytics.track(.groupedProductLinkedProductsDeleteButtonTapped)
-            self?.deleteProduct(model)
+            guard let self = self else { return }
+            ServiceLocator.analytics.track(self.deleteButtonTappedEvent)
```

Comment:
> Are we reusing this data source with different view controllers, and therefore we want to be able to send different events to tracks depending on the context?

Correct. This data source will be used by `LinkedProductsListSelectorViewController`, which is now generic and used in two points of the app: Grouped Products and Linked Products. Since there are two points where the list selector will be presented, we need to track different events, with different names.

> Because if that's the case, and building on the previous comment, maybe we could think about the parameters we pass to this class in a way that we provide an abstraction that models/describes the context?

Do you mean to add a `struct` to be passed to the `init` method which contains only the events that we want to track (in this case, just one)?

Reply:
> Do you mean to add a struct to be passed to the init method which contains only the events that we want to track (in this case, just one)?

I was wondering if there is anything else (logic, or other configuration) that is shared between the view controllers that share this list selector. But it seems like it's only one VC (LinkedProductsListSelectorViewController) so it might not be worth complicating things more.